src/Integration.Vsix/SonarLintDaemon/CFamily.cs

```diff
@@ -0,0 +1,471 @@
+                {
+                    default:
+                        throw new Exception("Unsupported CompileAs: " + value);
+                    case "Default":
+                        // Compile files with extensions ".cpp", ".cxx" and ".cc" as Cpp and files with extension ".c" as C
+                        if (path.ToLowerInvariant().EndsWith(".cpp") || path.ToLowerInvariant().EndsWith(".cxx") || path.ToLowerInvariant().EndsWith(".cc"))
+                        {
+                            sqLanguage = CPP_LANGUAGE_KEY;
+                        }
+                        else if (path.ToLowerInvariant().EndsWith(".c"))
+                        {
+                            sqLanguage = C_LANGUAGE_KEY;
+                        }
+                        else
+                        {
+                            sqLanguage = null;
+                        }
+                        return "";
+                    case "CompileAsC":
+                        sqLanguage = "c";
+                        return "/TC";
+                    case "CompileAsCpp":
+                        sqLanguage = CPP_LANGUAGE_KEY;
+                        return "/TP";
+                }
+            }
+
+            internal static string ConvertCompileAsManaged(string value)
+            {
+                switch (value)
+                {
+                    default:
+                        throw new Exception("Unsupported CompileAsManaged: " + value);
+                    case "":
+                    case "false":
+                        return "";
+                    case "true":
+                        return "/clr";
+                    case "Pure":
+                        return "/clr:pure";
+                    case "Safe":
+                        return "/clr:safe";
+                }
+            }
+
+            internal static string ConvertRuntimeLibrary(string value)
+            {
+                switch (value)
+                {
+                    default:
+                        throw new Exception("Unsupported RuntimeLibrary: " + value);
+                    case "MultiThreaded":
+                        return "/MT"; // defines macro "_MT"
+                    case "MultiThreadedDebug":
+                        return "/MTd"; // defines macros "_MT" and "_DEBUG"
+                    case "MultiThreadedDLL":
+                    case "MultiThreadedDll":
+                        return "/MD"; // defines macros "_MT" and "_DLL"
+                    case "MultiThreadedDebugDLL":
+                    case "MultiThreadedDebugDll":
+                        return "/MDd"; // defines macros "_MT", "_DLL" and "_DEBUG"
+                }
+            }
+
+            internal static string ConvertEnableEnhancedInstructionSet(string value)
+            {
+                switch (value)
+                {
+                    default:
+                        throw new Exception("Unsupported EnableEnhancedInstructionSet: " + value);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') 'System.Exception' should not be thrown by user code. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=csharpsquid%3AS112)